app/lib/redcarpet/render/html_rouge.rb

```diff
@@ -5,11 +5,31 @@ module Render
     class HTMLRouge < HTML
       include Rouge::Plugins::Redcarpet
 
-      # Rouge requires the hint language to be lower case, by overriding this
-      # method we can allow the hint language to be specified with other casings
-      # eg. `Ada` instead of `ada`
-      def block_code(code, language)
-        super(code, language.to_s.downcase)
+      # overrided method to add line number support
+      def block_code(code, language, opts = {})
```

Comment:
is this going to impact code blocks in the markdown around the website. I'm asking this as `HTMLRouge` is used both in comments and articles.

Should we add specific tests for this?

Reply:
Based on the changes I've made, for default markdown cases, it should call the function correct. I've extracted the original code from [redcarpet.rb](https://github.com/rouge-ruby/rouge/blob/master/lib/rouge/plugins/redcarpet.rb) in the original repo, and made a few adjustments.

This allows `Formatters::HTMLLegacy` to take in additional parameters for line number rendering.

If we need to add tests cases for this, where should I add them in?

Reply:
I'm cceing @maestromac which probably has insight about this

Reply:
This looks like a safe change as long as it passes all of our `spec/labor/markdown_parser_spec.rb` specs. @colbymelvin could you please add new tests to verify the functionality of this code? it can be added to `markdown_parser_spec.rb`

Reply:
Sure, will do that.

Reply:
@maestromac may I know what kind of test I should be adding into the `markdown_parser.spec.rb`? The `MarkdownParser` class uses `HTMLRouge` as it's renderer, but it passes it into `Redcarpet::Markdown` to render the markdown. But I'm using the HTMLRogue class directly to render the file itself.

Am I supposed to use `HTMLRogue` directly in the test to check it renders properly, or use the MarkdownParser class? I'm not exactly sure whether the class will call the `block_code` function.